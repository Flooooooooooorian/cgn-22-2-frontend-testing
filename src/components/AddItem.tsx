import {useState} from "react";

type AddItemProps = {
    addItem: (name:string) => void
}

export default function AddItem(props: AddItemProps) {

    const [name, setName] = useState<string>("")

    return (
        <>
            <input placeholder={"Name"} value={name} onChange={(event) => {setName(event.target.value)}}/>
            <button onClick={() => {props.addItem(name)}}>Save</button>
        </>
    )
}
