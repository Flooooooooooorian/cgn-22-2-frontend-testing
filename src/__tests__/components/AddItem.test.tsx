import {fireEvent, render, screen} from "@testing-library/react";
import AddItem from "../../components/AddItem";


describe("AddItem Tests", () => {

    test("AddItem Snapshot", () => {
        //GIVEN
        const mockAddItem = jest.fn();

        //WHEN
        const view = render(<AddItem addItem={mockAddItem}/>)

        //THEN
        expect(view.asFragment()).toMatchSnapshot();
    })

    test("AddItem submit test", () => {
        //GIVEN
        const mockAddItem = jest.fn();

        //WHEN
        const view = render(<AddItem addItem={mockAddItem}/>)

        //THEN
        fireEvent.input(screen.getByPlaceholderText("Name"), {target: {value: "TestInput"}})
        expect(view.asFragment()).toMatchSnapshot();

        fireEvent.click(screen.getByText("Save"))

        expect(mockAddItem).toBeCalledWith("TestInput")

    })
})
