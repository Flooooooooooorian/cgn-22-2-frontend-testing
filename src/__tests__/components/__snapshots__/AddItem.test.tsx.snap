// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItem Tests AddItem Snapshot 1`] = `
<DocumentFragment>
  <input
    placeholder="Name"
    value=""
  />
  <button>
    Save
  </button>
</DocumentFragment>
`;

exports[`AddItem Tests AddItem submit test 1`] = `
<DocumentFragment>
  <input
    placeholder="Name"
    value="TestInput"
  />
  <button>
    Save
  </button>
</DocumentFragment>
`;
